partner_firstname/models.py

```diff
@@ -155,7 +155,12 @@ def _get_inverse_name(self, name, is_company=False):
     def _inverse_name(self):
         """Try to revert the effect of :meth:`._compute_name`."""
         parts = self._get_inverse_name(self.name, self.is_company)
-        self.lastname, self.firstname = parts["lastname"], parts["firstname"]
+        values = {
+            "lastname": parts["lastname"],
+            "firstname": parts["firstname"]
+        }
+        self.write(values)
```

Comment:
That's all very nice, but misses the point entirely. If you `write` in an onchange handler, the user can't undo this easily by simply cancelling the current edit action. An onchange handler (and compute functions btw) never should have side effects, because you'll *always* run into situations where they fall on your feet.

As you write yourself, this belongs into a *install hook and/or migration scripts.